README-dev.md

```diff
@@ -6,36 +6,61 @@ This README is for SDK developers. If you are a plugin developer please refer to
 
 The artifact produced by this repository is a set of Python distributions that make up the SDK.
 
-# Development process
+## Background
 
-This repository is going through a lot of changes. It is being migrated to GitHub and open sourced. The development process will change throughout this process so please refer back to this README regularly to understand what the current development process is.
+There are two parts of the SDK that are important to think about separately since they have slightly different workflows.
 
-At a very high level, our development process usually looks like this:
+1. The `tools` package is the SDK's CLI. This aids in plugin development, testing, and distribution.
+2. `common`, `libs`, and `platform` contain what are collectively called the "wrappers". These are vanilla Python classes
+that abstract the Virtualization API protobuf messages (published by app-gate) away from plugin developers. These expose
+the API plugin developers write against.
+
+All dependencies of a plugin must be packaged with the plugin including the protobuf messages (`dvp-api`) and the wrappers.
+This is done automatically by `dvp build`.
 
-1. Make changes to SDK and appgate code. Test these changes manually. Iterate on this until you have everything working.
-2. Publish a development build of the SDK to artifactory.
-3. Update the version of the SDK specified in the app gate.
-4. Publish a review for SDK code, and also publish a "provisional" review of appgate code. Address any feedback.
-5. Push the SDK code and publish new SDK builds to our internal servers.
-6. Finalize your appgate review.
-7. Push the appgate changes
+This is what causes the slightly different workflows in development. Changes to `tools` are completely isolated from the
+Delphix Engine and wrappers changes only impact the plugin build.
 
-Not every type of change requires every step.
+## Development process
+
+At a very high level, our development process usually looks like this:
+
+1. Create a fork of the delphix/virtualization-sdk repository.
+2. Clone the forked repository.
+3. Make changes to SDK code. Test these changes manually and with unit tests. Iterate on this until you have everything working.
+4. Commit your changes and build all Python package distributions. Make sure the version number of the packages is updated appropriately.
+5. Publish Python distributions to artifactory. 
```

Comment:
Is this still true?

Reply:
Is it better to push to a branch in your fork and pass in the blackbox parameters?

Reply:
Yes. Ideally, we would push to a branch in your fork and pass point blackbox to the branch. However, we can't do it until blackbox can build SDK from source using the new Python build (blackbox currently relies on the old Gradle build). Until then, we have to point blackbox to the SDK version in artifactory.